splash-reward-distributor/src/indexer.rs

```diff
@@ -0,0 +1,168 @@
+use std::hash::Hash;
+use std::{fmt::Display, sync::Arc};
+
+use bloom_offchain::execution_engine::bundled::Bundled;
+use cml_crypto::Ed25519KeyHash;
+use rocksdb::TransactionDB;
+use serde::{de::DeserializeOwned, Serialize};
+use spectrum_offchain::domain::Stable;
+use spectrum_offchain::domain::{
+    event::{AnyMod, Confirmed, Predicted, Traced},
+    EntitySnapshot,
+};
+
+use crate::onchain::{
+    auth_manager::AuthManager, buffer_wallet::BufferWallet, harvest_order::HarvestOrder, smart_farm::Gauge,
+};
+
+#[async_trait::async_trait]
+pub trait OnChainIndex<Bearer> {
+    async fn read<T>(&self) -> Option<AnyMod<Bundled<T, Bearer>>>
+    where
+        T: unique_ids::UniqueId + EntitySnapshot + Send;
```

Comment:
How would I query specific gauge with this api?

Reply:
Sorry, that should be `read<T>(&self, id: T::StableId)...`